graph.py

```diff
@@ -9,12 +9,47 @@
     PACKAGE = 'networkx>=2.2'
     log.error(f"Package {PACKAGE} not found. Please 'pip install {PACKAGE}'")
 
+from itertools import combinations
+
+class Filter:
+    def __init__(self, filter_type, filter_function):
+        self._filter_type = filter_type
+        self._filter_fun = filter_function
+
+    def run(self,value, items):
+        if isinstance(value, self._filter_type):
+            fun0 = self._filter_fun(value)
+            return filter(fun0, items)
+        else:
+            raise TypeError(f"Expected type: {self._filter_type}")
+
+
 
 class KytosGraph:
     """Class responsible for the graph generation."""
 
     def __init__(self):
         self.graph = nx.Graph()
+        self._filter_fun_dict = {}
```

Comment:
Filter function dictionary contains a metric name as a key and its associated filter object as a value.